addon/mixins/adapter-fetch.js

```diff
@@ -0,0 +1,142 @@
+import Ember from 'ember';
+import fetch from 'fetch';
+
+const { assign, RSVP } = Ember;
+
+/**
+ * Helper function that turns the data/body of a request into a query param string.
+ * @param {Object} queryParamsObject
+ * @returns {String}
+ */
+export function serialiazeQueryParams(queryParamsObject) {
+  return Object.keys(queryParamsObject).map((key) => {
+    return `${encodeURIComponent(key)}=${encodeURIComponent(queryParamsObject[key])}`;
+  }).join('&');
+}
+
+/**
+ * Helper function to create a plain object from the response's Headers.
+ * Consumed by the adapter's `handleResponse`.
+ * @param {Headers} headers
+ * @returns {Object}
+ */
+export function headersToObject(headers) {
+  let headersObject = {};
+  headers.forEach((value, key) => {
+    headersObject[key] = value;
+  });
+  return headersObject;
+}
+/**
+ * Helper function that translates the options passed to `jQuery.ajax` into a format that `fetch` expects.
+ * @param {Object} options
+ */
+export function mungOptionsForFetch(_options) {
+  const options = assign({
+    credentials: 'same-origin',
+  }, _options);
+
+  options.method = options.type;
+
+  // GET and HEAD requests can't have a `body`
+  if (options.data && Object.keys(options.data).length) {
+    if (options.method === 'GET' || options.method === 'HEAD') {
+      options.url += `?${serialiazeQueryParams(_options.data)}`;
```

Comment:
I've been playing in the console, this snippet seems to work well for arbitrary nested objects:
```
function serialiazeQueryParams(queryParamsObject, prefix = null) {
  return Object.keys(queryParamsObject).sort().map((key) => {
    let value = queryParamsObject[key];
    if (prefix) {
      key = `${prefix}[${key}]`;
    }
    if (value !== null && typeof value === 'object') {
      return serialiazeQueryParams(value, key);
    } else {
      return `${encodeURIComponent(key)}=${encodeURIComponent(value)}`;
    }
  }).join('&');
}
```

I haven't checked how arrays should work with arrays.

For what is worth, this is the implementation of `$.param`, which is the thing used by `$.ajax`: https://github.com/jquery/jquery/blob/master/src/serialize.js#L17-L90